wpilibj/src/test/java/edu/wpi/first/wpilibj/event/BooleanEventTest.java

```diff
@@ -58,15 +59,89 @@ void testBinaryCompositionLoopSemantics() {
 
     assertEquals(1, counter1.get());
     assertEquals(1, counter2.get());
+
+    bool2.set(false);
+    loop1.poll();
+
+    assertEquals(2, counter1.get());
+    assertEquals(1, counter2.get());
+
+    loop2.poll();
+
+    assertEquals(2, counter1.get());
+    assertEquals(1, counter2.get());
+
+    loop1.poll();
+
+    assertEquals(2, counter1.get());
+    assertEquals(1, counter2.get());
+
+    bool2.set(true);
+    loop2.poll();
+
+    assertEquals(2, counter1.get());
+    assertEquals(1, counter2.get());
+
+    loop1.poll();
+
+    assertEquals(3, counter1.get());
+    assertEquals(1, counter2.get());
+
+    bool1.set(false);
+    loop2.poll();
+
+    assertEquals(3, counter1.get());
+    assertEquals(1, counter2.get());
+
+    loop1.poll();
+
+    assertEquals(3, counter1.get());
+    assertEquals(1, counter2.get());
+
+    loop2.poll();
+
+    assertEquals(3, counter1.get());
+    assertEquals(1, counter2.get());
+  }
+
+  /**
+   * When a BooleanEvent is constructed, an action is bound to the event loop to update an internal
+   * state variable. This state variable is checked during loop polls to determine whether or not to
+   * execute an action. If a condition is changed during the loop poll but before the state variable
+   * gets updated, the changed condition is used for the state variable.
```

Comment:
This comment mostly explains details about BooleanEvent internals that aren't entirely relevant to the test, and nearly no explanation about what's actually the behavior/case being tested here.

The behavior/case being tested should be indicated by the function name, optionally with a short API doc elaborating a bit more (the comments on `testMidLoopBooleanChange`, `testEdgeReuse` and `testEdgeReconstruct` are great!); if test flow is complex, inline comments should explain how that behavior/case is tested and why the assertions make sense.